### PR TITLE
Add article "Creating or Porting an Application to NuttX - Part 1"

### PR DESCRIPTION
**Pull request by tmedicci**
This is the first article of a series that explores NuttX's build system and how an application can be created and/or compiled for NuttX.

### CODE REVIEW

File: content/blog/nuttx-adding-porting-an-app/index.md [hunk 1]
```diff
@@ -0,0 +1,471 @@
+---
+title: "Creating or Porting an Application to NuttX - Part 1"
+date: 2024-08-06T08:00:00-03:00
+tags: ["NuttX", "Apache", "ESP32", "POSIX", "Linux", "Tutorial"]
+showAuthor: false
+authors:
+    - "tiago-medicci"
+---
```
Comment (pedrominatel) [inline content/blog/nuttx-adding-porting-an-app/index.md:8]: If this is part 1, I suggest you create it as a series. Please add the following:

```
series: ["Porting-to-NuttX"]
series_order: 1
```

Comment (tmedicci) [reply content/blog/nuttx-adding-porting-an-app/index.md:10]: Of course, I updated and rebased the article!

File: content/blog/nuttx-adding-porting-an-app/index.md [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+CFLAGS += ${DEFINE_PREFIX}MBEDTLS_CONFIG_FILE="<mbedtls/mbedtls_config.h>"
+
+CXXFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/include
+CXXFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/mbedtls/include
+CXXFLAGS += ${DEFINE_PREFIX}MBEDTLS_CONFIG_FILE="<mbedtls/mbedtls_config.h>"
+
+endif
+{{< /highlight >}}
+
+## Conclusion? (not yet!)
```
Comment (pedrominatel) [inline content/blog/nuttx-adding-porting-an-app/index.md:451]: Maybe you can use "to be continued"
## Conclusion (to be continued)

**Review by pedrominatel — COMMENTED** [review]
Please find my comments, @tmedicci.
Thank you!

File: content/blog/nuttx-adding-porting-an-app/index.md [hunk 3]
```diff
@@ -0,0 +1,473 @@
+---
+title: "Building Applications on NuttX: Understanding the Build System"
+date: 2024-08-12T08:00:00-03:00
```
Comment (pedrominatel) [inline content/blog/nuttx-adding-porting-an-app/index.md:3]: Change to a recent date.

File: content/blog/nuttx-adding-porting-an-app/index.md [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+
+Giving a glimpse about it: have you ever thought about how cool it would be to turn an ESP32 into a low-cost, high-fidelity sound server? Well, we might need `RTP Tools` for that :wink:.
+
+## Compiling an Application in NuttX
+
+Applications in NuttX are separated from the kernel source code of the operating system. Similarly to other operating systems, they interact with the NuttX kernel through public APIs compatible with the POSIX standard. It's worth noting, however, that applications need to use the NuttX build system to be integrated into the final firmware. Thus, the Apache NuttX project also provides the [`nuttx-apps`](https://github.com/apache/nuttx-apps) repository, which is a collection of applications available for NuttX. The applications in this repository are public and can be used by any user. NuttX also provides means to compile applications outside this repository, allowing proprietary applications (not public) to be integrated into the system.
```
[the remark this answers is not in the document]
Comment (tmedicci) [reply content/blog/nuttx-adding-porting-an-app/index.md]: Done!

File: content/blog/nuttx-adding-porting-an-app/index.md (hunk 2 above)
Comment (tmedicci) [reply content/blog/nuttx-adding-porting-an-app/index.md]: Done!

File: content/blog/nuttx-adding-porting-an-app/index.md (hunk 3 above)
Comment (tmedicci) [reply content/blog/nuttx-adding-porting-an-app/index.md]: Done!

**Comment by tmedicci** [comment]
@pedrominatel and @f-hollow , please note that the failing CI check refers to a piece of code (not a link, by definition) and the URL is being composed (it isn't the final URL):
```
MBEDTLS_URL ?= "https://github.com/ARMmbed/mbedtls/archive"

MBEDTLS_VERSION = $(patsubst "%",%,$(strip $(CONFIG_MBEDTLS_VERSION)))
MBEDTLS_ZIP = v$(MBEDTLS_VERSION).zip

$(MBEDTLS_ZIP):
	@echo "Downloading: $(MBEDTLS_URL)/$(MBEDTLS_ZIP)"
	$(Q) curl -O -L $(MBEDTLS_URL)/$(MBEDTLS_ZIP)
```

I couldn't find a way to ignore checking code pieces, though.

**Comment by pedrominatel** [comment]
Can you kindly add this link to the `.lycheeignore` file?

**Review by pedrominatel — APPROVED** [review]

**Pull request merged**